Sources/SwiftParser/Availability.swift

```diff
@@ -286,24 +307,29 @@ extension Parser {
         } else {
           trailingComponents.append(versionComponent)
         }
-      }
 
-      var unexpectedTrailingComponents: RawUnexpectedNodesSyntax?
-
-      if !trailingComponents.isEmpty {
-        unexpectedTrailingComponents = RawUnexpectedNodesSyntax(elements: trailingComponents.compactMap { $0.as(RawSyntax.self) }, arena: self.arena)
+        if version.isMissing {
+          break
+        }
       }
 
+      let unexpectedTrailingComponents = RawUnexpectedNodesSyntax(elements: trailingComponents.map(RawSyntax.init), arena: self.arena)
```

Comment:
I would do the following

```suggestion
      let unexpectedTrailingComponents = RawUnexpectedNodesSyntax(elements: trailingComponents.map(RawSyntax.init), arena: self.arena)
```

This uses the following initializer

https://github.com/apple/swift-syntax/blob/1555299f9e1fac77cf92677dae4dc8b46e9aa3d0/Sources/SwiftParser/SyntaxUtils.swift#L39-L43

It has two advantages:
1. If `trailingComponents` is empty, it doesn’t allocate a `RawUnexpectedNodesSyntax`, reducing the memory footprint (remember that raw syntax nodes are allocated in a `SyntaxArena` and thus don’t get freed until the entire arena gets freed)
2. It’s also shorter 😉

Reply:
Oh, I understand. It's more efficient to return nil if the array is empty 😃